src/test/java/bio/terra/service/snapshotbuilder/SnapshotBuilderServiceTest.java

```diff
@@ -94,26 +97,75 @@ void enumerateSnapshotRequestsByDatasetId() {
   @ParameterizedTest
   @EnumSource(CloudPlatform.class)
   public void getConceptChildren(CloudPlatform cloudPlatform) {
-    Dataset dataset =
-        new Dataset(new DatasetSummary().cloudPlatform(cloudPlatform))
-            .projectResource(new GoogleProjectResource().googleProjectId("project123"))
-            .name("dataset123")
-            .id(UUID.randomUUID());
+    Dataset dataset = makeDataset(cloudPlatform);
+    when(datasetService.retrieve(dataset.getId())).thenReturn(dataset);
+    var concepts = List.of(new SnapshotBuilderConcept().name("concept1").id(1));
+    mockRunQueryForConcepts(cloudPlatform, concepts, dataset);
+    var response = snapshotBuilderService.getConceptChildren(dataset.getId(), 1, TEST_USER);
+    assertThat(
+        "getConceptChildren returns the expected response",
+        response.getResult(),
+        equalTo(concepts));
+  }
+
+  @ParameterizedTest
+  @EnumSource(CloudPlatform.class)
+  void searchConcepts(CloudPlatform cloudPlatform) {
+    Dataset dataset = makeDataset(cloudPlatform);
     when(datasetService.retrieve(dataset.getId())).thenReturn(dataset);
-    CloudPlatformWrapper cloudPlatformWrapper = CloudPlatformWrapper.of(cloudPlatform);
     var concepts = List.of(new SnapshotBuilderConcept().name("concept1").id(1));
+    mockRunQueryForConcepts(cloudPlatform, concepts, dataset);
+    var response =
+        snapshotBuilderService.searchConcepts(dataset.getId(), "condition", "cancer", TEST_USER);
+    assertThat(
+        "searchConcepts returns the expected response", response.getResult(), equalTo(concepts));
+  }
+
+  @ParameterizedTest
+  @EnumSource(CloudPlatform.class)
+  void getTableNameGenerator(CloudPlatform cloudPlatform) {
+    Dataset dataset = makeDataset(cloudPlatform);
+    DatasetModel model = makeDatasetModel();
+    String tableName = "table";
+    String dataSource = "dataSource";
+    CloudPlatformWrapper wrapper = CloudPlatformWrapper.of(cloudPlatform);
+    if (wrapper.isAzure()) {
+      when(datasetService.getOrCreateExternalAzureDataSource(dataset, TEST_USER))
+          .thenReturn(dataSource);
+      assertThat(
+          "getTableNameGenerator returns a TableNameGenerator that generates the expected table name",
+          snapshotBuilderService.getTableNameGenerator(TEST_USER, dataset).generate(tableName),
+          is(SynapseVisitor.azureTableName(dataSource).generate(tableName)));
```

Comment:
This is not good I know there must be some better way to test this method

Reply:
I mentioned this in person, but while looking at this I noticed that we have both `SnapshotBuilderService.getTableNameGenerator(AuthenticatedUserRequest, Dataset)` and `getTableNameGenerator(Dataset, AuthenticatedUserRequest)` which are doing the same thing. So one can be removed and maybe these tests can be removed too.

Reply:
I agree that this way of testing isn't great but there may be no better way as the references to synapse and bigquery are hardcoded in `getTableNameGenerator()` and can't easily be mocked.

My guess is that this path may already be covered by other tests that are generating SQL text. If so, I think it's OK to not have a separate unit test for this method.